lib/pg_rls/tenant.rb

```diff
@@ -18,7 +18,7 @@ def switch(resource)
       attr_reader :tenant
 
       def fetch
-        @fetch ||= tenant.find_by_tenant_id(
+        PgRls.main_model.find_by_tenant_id(
```

Comment:
Good catch, don't know how I missed that, probably something I change during some upgrades 😞